src/libraries/System.Security.Claims/src/System/Security/Claims/ClaimsIdentity.cs

```diff
@@ -933,5 +936,48 @@ protected virtual void GetObjectData(SerializationInfo info, StreamingContext co
         {
             throw new PlatformNotSupportedException();
         }
+
+        internal string DebuggerToString()
+        {
+            // DebuggerDisplayAttribute is inherited. Use virtual members instead of private fields to gather data.
+            int claimsCount = 0;
+            foreach (var item in Claims)
+            {
+                claimsCount++;
+            }
+
+            return $"Identity Name = {Name ?? "(null)"}, IsAuthenticated = {IsAuthenticated}, Claims Count = {claimsCount}";
+        }
+
+        private sealed class ClaimsIdentityDebugProxy
```

Comment:
What are the benefits of using custom `[DebuggerTypeProxy]` in this particular example? Showing only selected properties?

Reply:
It shows only public properties.